Lesson4_wordbreak.cpp

```diff
@@ -0,0 +1,136 @@
+/*Word Break
+Given an input string and a dictionary of words, find out if the input string can be segmented into a space-separated sequence of dictionary words.
+For example, consider the following dictionary: { pear, salmon, foot, prints, footprints, leave, you, sun, girl, enjoy },
+
+Examples:
+Given the string “youenjoy”, 
+Output: True (The string can be segmented as “you enjoy”)
+
+Input: “youleavefootprints”,
+Output: True (The string can be segmented as “you leave footprints” or “you leave foot prints”)
+
+Input:salmonenjoyapples
+Output: False
+*/
+#include<iostream>
+#include<cstring>
+#include<vector>
+using namespace std;
+bool findword(string word,vector<string> dict)
+{
+	for(int i=0;i<dict.size();++i)
+	{
+		if(!(dict[i].compare(word)))
+			return true;
+	}
+	return false;
+	
+}
+bool wordbreak(string str,vector<string> dict)
+{
+	int strlen,dictlen;
+	strlen=str.size();
+	dictlen=dict.size();
+	if(strlen==0)
+		return true;
+	if(dictlen==0)
+		return false;
+	int dp[strlen+1]={0};
```

Comment:
since you are using these values as binary 0/1, bool[] would be a more clear choice than int[].  Also, what does dp stand for? It might be better to use a little bit clearer name for this

Reply:
looks like your comment on line 40 got moved to the wrong place!